src/pages/contribute/components/Contribute.js

```diff
@@ -124,57 +123,60 @@ const Comp = ({
 
   return (
     <div className="tile is-child">
-      {(!web3 || effectiveBalance === 0) && showwelcome ? <WelcomeModal /> : null}
+      {!web3 && showwelcome ? <WelcomeModal /> : null}
```

Comment:
```suggestion
      {(!web3 || cstkBalance === '0') && showwelcome || ? <WelcomeModal /> : null}
```

Maybe? As I don't think it makes sense to show the welcome modal with all the steps once the user has already its CSTK tokens.